app/controllers/application_pets_controller.rb

```diff
@@ -1,8 +1,18 @@
 class ApplicationPetsController < ApplicationController
   def update
     application_pet = ApplicationPet.find_by(application_id: params[:application_id], pet_id: params[:pet_id])
-    application_pet.update(approved: true)
-    application_pet.pet.update(status: "Pending")
-    redirect_to "/pets/#{application_pet.pet.id}"
+    if params[:approval] == 'approve'
+      unless application_pet.pet.status == "Pending"
+        application_pet.update(approved: true)
+        application_pet.pet.update(status: "Pending")
+      else
+        flash[:notice] = "#{application_pet.pet.name} is already pending adoption."
```

Comment:
added this edge case to handle when two applications try to approve the same pet